src/relay/pass/enable_tensorrt.cc

```diff
@@ -0,0 +1,585 @@
+    {"expand_dims"},
+    {"squeeze"},
+    {"concatenate"},
+    {"nn.conv2d_transpose"},
+    {"transpose"},
+    {"reshape"},
+    {"nn.pad"},
+    {"sum"},
+    {"prod"},
+    {"max"},
+    {"min"},
+    {"mean"},
+    {"contrib.adaptive_max_pool2d"},
+    {"contrib.adaptive_avg_pool2d"}};
+
+// Ops which require TRT 5.1.5+
+static const std::unordered_set<std::string> trt_5_1_5_compatible_ops = {
+    {"clip"}, {"nn.leaky_relu"}, {"sin"},   {"cos"},
+    {"atan"}, {"ceil"},          {"floor"}, {"strided_slice"}};
+
+// Ops which require TRT 6.0.1+
+static const std::unordered_set<std::string> trt_6_0_1_compatible_ops = {
+    {"image.resize"}};
+
+bool TrtVersionGe(const std::tuple<int, int, int>& curr_version, int major,
+                  int minor, int patch) {
+  if (std::get<0>(curr_version) > major) return true;
+  if (std::get<0>(curr_version) == major && std::get<1>(curr_version) > minor)
+    return true;
+  if (std::get<0>(curr_version) == major &&
+      std::get<1>(curr_version) == minor && std::get<2>(curr_version) >= patch)
+    return true;
+  return false;
+}
+
+std::vector<int> GetShape(const Type& type) {
+  const auto* ttype = type.as<TensorTypeNode>();
+  CHECK(ttype);
+  std::vector<int> _shape;
+  for (size_t i = 0; i < ttype->shape.size(); ++i) {
+    auto* val = ttype->shape[i].as<IntImm>();
+    CHECK(val);
+    _shape.push_back(val->value);
+  }
+  return _shape;
+}
+
+class TrtChecker : public ExprVisitor {
+ public:
+  explicit TrtChecker(const std::tuple<int, int, int>& trt_version)
+      : trt_version_(trt_version) {
+    // Create set of compatible ops for this version.
+    trt_compatible_ops.insert(trt_base_compatible_ops.begin(),
+                              trt_base_compatible_ops.end());
+    if (TrtVersionGe(trt_version_, 5, 1, 5)) {
+      // Add TRT 5.1.5 ops to whitelist.
+      trt_compatible_ops.insert(trt_5_1_5_compatible_ops.begin(),
+                                trt_5_1_5_compatible_ops.end());
+    }
+    if (TrtVersionGe(trt_version_, 6, 0, 1)) {
+      // Add TRT 6.0.1 ops to whitelist.
+      trt_compatible_ops.insert(trt_6_0_1_compatible_ops.begin(),
+                                trt_6_0_1_compatible_ops.end());
+    }
+  }
+
+  void VisitExpr_(const VarNode* op) {
+    const auto* ttype = op->checked_type().as<TensorTypeNode>();
+    CHECK(ttype);
+    if (!ttype->dtype.is_float()) {
+      compatible_ = false;
+    }
+  }
+
+  void VisitExpr_(const CallNode* call) final {
+    const std::string op_name = (call->op.as<OpNode>())->name;
+    for (size_t i = 0; i < call->args.size(); ++i) {
+      // Workaround for check for transpose of weight.
+      if ((op_name == "nn.conv2d" || op_name == "nn.dense") && i == 1) {
+        if (auto* transpose = call->args[i].as<CallNode>()) {
+          if (transpose->op.as<OpNode>()->name == "transpose") {
+            if (!transpose->args[0].as<VarNode>()) {
+              compatible_ = false;
+              LOG(INFO) << op_name
```

Comment:
We probably want to use DLOG.

Reply:
I would prefer to use LOG. It is useful to easily see why TRT couldn't be enabled for a model and since this step is done in compilation stage we don't have the option to enable debug logging.